Framework/Algorithms/src/Rebin.cpp

```diff
@@ -263,8 +260,7 @@ void Rebin::init() {
                   "is linear. Power must be between 0 and 1.");
 
   declareProperty(
-      PropertyNames::BINMODE, binningModeNames[size_t(BinningMode::DEFAULT)],
-      std::make_shared<Mantid::Kernel::StringListValidator>(binningModeNames),
+      std::make_unique<EnumeratedStringProperty<BinningMode, &binningModeNames>>(PropertyNames::BINMODE),
```

Comment:
I'm not sure why the standard way to add properties is with `std::shared_ptr` rather than `std::unique_ptr`. Is this right or wrong?